src/aggregated/circulating_supply/lockup_types.rs

```diff
@@ -1,3 +1,4 @@
+#![allow(clippy::all)]
```

Comment:
is it really necessary here?

Reply:
This file gives ~10 warnings. I don't want to touch copy-pasted code (and, to be honest, I don't know how to fix some of these warnings properly 😄)

Reply:
Could you share some of them so we can have a look?

cc @frol

Reply:
11, and they all go from `construct_uint!`

```
warning: manual `RangeInclusive::contains` implementation
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = note: `#[warn(clippy::manual_range_contains)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#manual_range_contains
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: manual implementation of an assign operation
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = note: `#[warn(clippy::assign_op_pattern)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assign_op_pattern
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: manual implementation of an assign operation
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assign_op_pattern
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: manual implementation of an assign operation
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assign_op_pattern
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: manual implementation of an assign operation
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assign_op_pattern
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of `offset` with a `usize` casted to an `isize`
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = note: `#[warn(clippy::ptr_offset_with_cast)]` on by default
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_offset_with_cast
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of `offset` with a `usize` casted to an `isize`
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_offset_with_cast
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of `offset` with a `usize` casted to an `isize`
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_offset_with_cast
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of `offset` with a `usize` casted to an `isize`
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_offset_with_cast
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: use of `offset` with a `usize` casted to an `isize`
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#ptr_offset_with_cast
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: manual implementation of an assign operation
  --> src/aggregated/circulating_supply/lockup_types.rs:12:1
   |
12 | / construct_uint! {
13 | |     /// 256-bit unsigned integer.
14 | |     pub struct U256(4);
15 | | }
   | |_^
   |
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#assign_op_pattern
   = note: this warning originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

warning: 11 warnings emitted

```

Reply:
The warnings come from uint macro, so we cannot really do anything about it. Let's at least narrow the `allow` list down.

```suggestion
#![allow(clippy::ptr_offset_with_cast, clippy::assign_op_pattern)]
```